### ISSUE TITLE
security/acme-client: Failing to get certificates from Gandi LiveDNS

### ISSUE THREAD

**Issue by FreakyBigFoot** [opnsense/plugins#3526]
**The bug**
I'm not able to renew a certificate when using the Challenge Type of DNS-01 via Gandi LiveDNS. 
```
2023-08-06T21:51:18-07:00` | acme.sh | [Sun Aug  6 21:51:18 PDT 2023] See: https://github.com/acmesh-official/acme.sh/wiki/How-to-debug-acme.sh
-- | -- | --
2023-08-06T21:51:18-07:00 | acme.sh | [Sun Aug  6 21:51:18 PDT 2023] Please add '--debug' or '--log' to check more details.
2023-08-06T21:51:18-07:00 | acme.sh | [Sun Aug  6 21:51:18 PDT 2023] Error add txt for domain:_acme-challenge.mydomain.xyz
2023-08-06T21:51:16-07:00 | acme.sh | [Sun Aug  6 21:51:16 PDT 2023] Adding txt value:  yehi3EBS36zpUVEDVrs4QEmJYWLldyXVtISx2MDIgGg for domain:   _acme-challenge.mydomain.xyz
2023-08-06T21:51:16-07:00 | acme.sh | [Sun Aug  6 21:51:16 PDT 2023] Getting webroot for domain='*.mydomain.xyz'
2023-08-06T21:51:16-07:00 | acme.sh | [Sun Aug  6 21:51:16 PDT 2023] Getting webroot for domain='mydomain.xyz'
2023-08-06T21:51:15-07:00 | acme.sh | [Sun Aug  6 21:51:15 PDT 2023] Getting domain auth token for each domain
2023-08-06T21:51:15-07:00 | acme.sh | [Sun Aug  6 21:51:15 PDT 2023] Multi domain='DNS:mydomain.xyz,DNS:*.mydomain.xyz'
2023-08-06T21:51:15-07:00 | acme.sh | [Sun Aug  6 21:51:15 PDT 2023] Using CA: https://acme-v02.api.letsencrypt.org/directory
```

I've updated my API key and even attempted to setup timeouts. The DNS txt record is never created. I'm not sure how to troubleshoot this more. Any help is appreciated.

OPNsense 23.7
AMD Ryzen 7 5800U
Network Intel® i226

**Comment by fraenki** [comment]
Please enable debug logging on the Acme-Client settings page and try again. This may reveal why it is failing.

**Comment by xer0x** [comment]
@FreakyBigFoot I have a hunch that switching your `/var/etc/acme-client/accounts/**/account.conf` to use `GANDI_LIVEDNS_TOKEN` instead of `GANDI_LIVEDNS_KEY` will fix your problem.

Gandi switched to using a Personal Access Token for their [authentication](https://api.gandi.net/docs/authentication/).

The ACME plugin UI looks like it might set the `GANDI_LIVEDNS_KEY`
The plugin looks like it has a little extra logic to handle both: https://github.com/acmesh-official/acme.sh/blob/master/dnsapi/dns_gandi_livedns.sh#L23

**Comment by FreakyBigFoot** [comment]
Hey @xer0x  Thank you very much! That worked. @fraenki So sorry.. I never saw this response. It's been a crazy month or 2. 

I had to also paste my key in there. I wonder how we can get this updated in the UI to use the newer system?

**Comment by fraenki** [comment]
Support for Personal Access Tokens will be available in os-acme-client 3.20.
